src/pages/functional-testing-framework/metadata.md

```diff
@@ -264,6 +268,154 @@ _Catalog/Data/CategoryData.xml_:
+    <requiredEntity type="company_relation">CompanyRelationData</requiredEntity>
+</entity>
+```
+
+- `type="company_relations"` points to the operation with `dataType=company_relations"` and `type="create"` in the _Metadata_ directory.
+- `<requiredEntity type="company_relation">CompanyRelationData</requiredEntity>` points to the data entity with `name="CompanyRelationData"` and `type="company_relation"`.
+
+_CompanyRelations/Data/CompanyIdsData.xml_:
+
+```xml
+<entity name="CompanyRelationData" type="company_relation">
+    <var key="company_id" entityType="company2" entityKey="id"/>
+</entity>
+```
+
+- `type="company_relation"` points to the operation with `dataType=company_relation"` and `type="create"` in the _Metadata_ directory.
+- The `company_id` will only get the `id` from the company with `type="company2"`
+
+_CompanyRelation/Metadata/CompanyIdsMeta.xml_:
+
+```xml
+<operation name="CreateCompanyRelation" dataType="company_relation" type="create" >
+    <field key="company_id">integer</field>
+</operation>
+```
+
+`<field key="company_id">integer</field>` will create an object with a key `company_id` with a value of type integer.
+```json
+{
+  "company_id": 1
+}
+```
+
+_CompanyRelation/Metadata/CompanyRelationsMeta.xml_:
+
+```xml
+<operation name="CreateCompanyRelations" dataType="company_relations" type="create" auth="adminOauth" url="/V1/company/{company.id}/relations" method="POST">
+    <contentType>application/json</contentType>
+    <array key="relations">
+        <value>company_relation</value>
+    </array>
+</operation>
+```
+
+As a result, the Functional Testing Framework sends a POST request with an array of company_ids in the body to the `https://example.com/rest/V1/company/:parentId/relations`.
+- `{company.id}` in the url comes from the test `<requiredEntity createDataKey="company1"/>` in EstablishOneCompanyRelationData that has `type="company"`
+- `<array key="relations">` will create an array with the key relations
+- `<value>company_relation</value>` will get the data from the operation with `dataType="company_relation"`
+```json
+{
+  "relations": [
+    {
+      "company_id":  1
+    }
+  ]
+}
+```
+
+##### Delete a company relation
+
+```xml
+<deleteData createDataKey="createCompanyRelation" stepKey="deleteCompanyRelation"/>
+```
+
+`createDataKey="createCompanyRelation"` will be the same as the createData `stepKey="createCompanyRelation"` in the test. This will call the delete operation that has the same `dataType="company_relations"` as the createData.
+
+```xml
+<operation name="DeleteCompanyRelations" dataType="company_relations" type="delete" auth="adminOauth" url="/V1/company/{company.id}/relations/{company2.id}" method="DELETE">
+    <contentType>application/json</contentType>
+</operation>
+```
+
+As a result, the Functional Testing Framework sends a DELETE request to the `https://example.com/rest/V1/company/:parentId/relations/:companyId`.
```

Comment:
> https://example.com/rest/V1/company/:parentId/relations/:companyId

change to "https://example.com/rest/V1/company/{company.id}/relations/{company2.id}"

Reply:
fixed